geode-core/src/main/java/org/apache/geode/internal/cache/CachePerfStats.java

```diff
@@ -1413,36 +1404,30 @@ public void endJob() {
     };
   }
 
-  public long getRegionClearCount() {
-    return stats.getLong(regionClearsId);
-  }
-
-  public long getBucketClearCount() {
-    return stats.getLong(bucketClearsId);
-  }
-
-  public long getPartitionedRegionClearLocalDuration() {
-    return stats.getLong(partitionedRegionClearLocalDurationId);
-  }
-
-  public long getPartitionedRegionClearTotalDuration() {
-    return stats.getLong(partitionedRegionClearTotalDurationId);
+  public long getClearCount() {
+    return stats.getLong(clearsId);
   }
 
-  public void incRegionClearCount() {
-    stats.incLong(regionClearsId, 1L);
+  public long startClear() {
+    long clearStartTime = getTime();
+    stats.incLong(clearsInProgressId, 1L);
+    return clearStartTime;
   }
 
-  public void incBucketClearCount() {
-    stats.incLong(bucketClearsId, 1L);
+  public long endClear(long startTime) {
+    long timeTaken = getTime() - startTime;
+    stats.incLong(clearTimeId, timeTaken);
```

Comment:
you should only call incLong(clearTimeId) if (clock.isEnabled())

Reply:
Done